### PR TITLE
Fixed bug related to apollo cache operations overlapping

### PR DESCRIPTION
**Pull request by innopoop**
[Task Description](https://www.notion.so/novainternal/Bug-popups-up-when-you-try-to-sign-up-with-an-already-existing-account-related-to-apollo-cache-e5852112a6c04ca6acf87650f9790856)

Other things taken care of in this PR:
1. Login was previously unabstracted (whoops), now it is abstracted
2. Took out check for if user is signed in and goes to /login and /signup (too complicated with the login from google sign up
3. Forward the user from program homepage to apply after authentication modal login
4. remove signedInIndicator and reload page after redirecting to show the user is logged in
5. Removed /my/applications from file structure since we are keeping applications on dashboard

### CODE REVIEW

File: src/components/Authentication/Signup.tsx [hunk 1]
```diff
@@ -58,10 +59,12 @@ const Signup = ({
                   }
                 );
               } else {
+                console.log(loading);
```
Comment (legitmaxwu) [inline src/components/Authentication/Signup.tsx:62]: remove console.log statement

File: src/pages/signup.tsx [hunk 2]
```diff
@@ -20,17 +18,8 @@ const BlobCircle = () => {
 };
 
 const SignUpPage: Page = () => {
-  const authorizationLevel = useAuthorizationLevel();
   const router = useRouter();
 
-  if (
```
Comment (legitmaxwu) [inline src/pages/signup.tsx:26]: I'll assume this is deleted for the login page as well?

**Review by legitmaxwu — CHANGES_REQUESTED** [review]
Good stuff--only minor comments. Could you leave a more detailed description in the PR as to what exactly you fixed (or link the related Notion task or something)? There's not enough information for me to understand why this PR is necessary and what I should be looking out for during code review.

**Comment by nova-admin** [comment]
#### 📝Changed routes:


| [**/login**](https://risc-2ukcvygh7-novaforgood.vercel.app/login) | [**/signup**](https://risc-2ukcvygh7-novaforgood.vercel.app/signup) |
|:-:|:-:|
|<a href="https://risc-2ukcvygh7-novaforgood.vercel.app/login">Screenshot of /login</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/a546Vr9hklbyveon-i6lvsjE-6qTcGDyFkNxlYs86aypmZhKT31u077xZAVKsZFUjrK3gZJqsziKOrpS9-wgvg.png">(view full size)</a> |<a href="https://risc-2ukcvygh7-novaforgood.vercel.app/signup">Screenshot of /signup</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/rrOuSJD56sH6U10_nmpnfN768SpNFzrmnuQ00Whwv1OQJVk2FDoX4tpOhyBBpilsybI8pQCWfjjv-hTN6ewTeg.png">(view full size)</a> |







Commit 58a527ef9e0a80348d53dc3c8dd6c325c6f40606 (https://risc-2ukcvygh7-novaforgood.vercel.app).

File: src/pages/program/[slug]/index.tsx [hunk 3]
```diff
@@ -42,6 +41,7 @@ const ReadOnlyHome = ({
   inProgram = false,
 }: DisplayProgramHomepageProps & { inProgram?: boolean }) => {
   const [authenticationModalOpen, setAuthenticationModalOpen] = useState(false);
+  const [role, setRole] = useState<"mentor" | "mentee">("mentor");
```
Comment (legitmaxwu) [inline src/pages/program/[slug]/index.tsx:44]: Unnecessary state variable. just do

`const role = ...` instead

**Review by legitmaxwu — APPROVED** [review]

**Pull request merged**